### PR TITLE
Added support for product quantities

### PR DESCRIPTION
**Pull request by andreheuer**
## Description

As I want to use pygrocy to add new items to the shopping list, I was missing the support to handle quantities of products. Therefore, I made following two changes:

1. Added support to QuantityUnit in the product data model. This is filled with quantity information of products.
2. Added the field qu_factor_purchase_to_stock of a product to get the default conversion rate for adding items to the shopping list.
3. Extended the add_product_to_shopping_list function to support a different quantity unit.
4. Extended tests to cover above changes (if applicable)

**Related issue (if applicable):** 
n/a

## Checklist
  - [x] The code change is tested and works locally.
  - [x] tests pass. **Your PR won't be merged unless tests pass**
  - [x] There is no commented out code in this PR

### CODE REVIEW

File: pygrocy/data_models/product.py [hunk 1]
```diff
@@ -47,6 +70,9 @@ def _init_empty(self):
         self._available_amount = None
         self._best_before_date = None
 
+        self._default_quantity_unit_purchase = None
```
Comment (SebRut) [inline pygrocy/data_models/product.py:73]: Could you add basic test assertions [over here](https://github.com/SebRut/pygrocy/blob/ffcb306a808ea4b202adc54985d33ad5d9eaa5c0/test/test_product.py#L25) for these fields please?

**Review by SebRut — CHANGES_REQUESTED** [review]

File: pygrocy/data_models/product.py (hunk 1 above)
Comment (andreheuer) [reply pygrocy/data_models/product.py:73]: Done. I have added some assertions (and forgot to commit file ;-))

**Review by SebRut — APPROVED** [review]

**Comment by SebRut** [comment]
@andreheuer hey, could you take a look at the formatting errors provided by black (https://github.com/psf/black) :)

**Comment by andreheuer** [comment]
@SebRut  I have reformatted the code with black. Hopefully, all is fine right now :-)

**Review by SebRut — APPROVED** [review]

**Comment by SebRut** [comment]
@andreheuer could we try to get this merged? Seems like the commits are not signed, is that something you could do?

**Pull request merged**